File: docs/obciazenie/connection.php
<?php

declare(strict_types=1);

/** Parametry tylko izolowanej, jawnie wskazanej bazy. Bez połączenia w walidacji. */
function benchmarkConnection(): array
{
    $db = getenv('BENCH_DB') ?: '';
    $host = getenv('PGHOST') ?: '';
    $port = getenv('PGPORT') ?: '';
    $user = getenv('PGUSER') ?: '';
    if (! preg_match('/\Akuking_bench_[a-z0-9_]{1,40}\z/', $db)
        || ! in_array($host, ['127.0.0.1', 'localhost'], true)
        || ! preg_match('/\A[0-9]{4,5}\z/', $port)
        || (int) $port < 1024 || (int) $port > 65535 || (int) $port === 5432
        || ! preg_match('/\A[a-z_][a-z0-9_]{0,62}\z/', $user)
        || strlen((string) getenv('PGHOSTADDR')) > 0
        || strlen((string) getenv('PGSERVICE')) > 0
        || strlen((string) getenv('PGSERVICEFILE')) > 0) {
        throw new InvalidArgumentException('Wskaż BENCH_DB kuking_bench_*, lokalny PGHOST, izolowany PGPORT (nie 5432) i PGUSER.');
    }

    return ["pgsql:host={$host};port={$port};dbname={$db}", $user, getenv('PGPASSWORD') ?: null];
}

File: docs/obciazenie/gobench/connection.go
package main

import (
	"net"
	"net/url"
	"os"
	"regexp"
	"strconv"
)

// Ta sama jawna granica co w PHP i build_scale.sh; brak domyślnego serwera/bazy.
func dsn(db string) string {
	host, port, user := os.Getenv("PGHOST"), os.Getenv("PGPORT"), os.Getenv("PGUSER")
	n, err := strconv.Atoi(port)
	if !regexp.MustCompile(`^kuking_bench_[a-z0-9_]{1,40}$`).MatchString(db) ||
		(host != "127.0.0.1" && host != "localhost") ||
		!regexp.MustCompile(`^[0-9]{4,5}$`).MatchString(port) || err != nil || n < 1024 || n > 65535 || n == 5432 ||
		!regexp.MustCompile(`^[a-z_][a-z0-9_]{0,62}$`).MatchString(user) ||
		os.Getenv("PGHOSTADDR") != "" || os.Getenv("PGSERVICE") != "" || os.Getenv("PGSERVICEFILE") != "" {
		panic("Wskaż BENCH_DB kuking_bench_*, lokalny PGHOST, izolowany PGPORT (nie 5432) i PGUSER.")
	}
	u := url.URL{Scheme: "postgres", Host: net.JoinHostPort(host, port), Path: "/" + db, User: url.User(user)}
	if password, ok := os.LookupEnv("PGPASSWORD"); ok {
		u.User = url.UserPassword(user, password)
	}
	return u.String()
}
